Xwt.XamMac/Xwt.Mac.CellViews/CompositeCell.cs

```diff
@@ -88,9 +88,23 @@ public void SetCurrentEventRow ()
 			source.SetCurrentEventRow (tablePosition.Position);
 		}
 
+		bool recalculatingHeight = false;
 		public void InvalidateRowHeight ()
 		{
-			source.InvalidateRowHeight (tablePosition.Position);
+			if (tablePosition != null && !recalculatingHeight) {
+				recalculatingHeight = true;
+				source.InvalidateRowHeight (tablePosition.Position);
+				recalculatingHeight = false;
+			}
+		}
+
+		public double GetRequiredHeightForWidth (double width)
+		{
+			Fill (false);
+			double height = 0;
+			foreach (var c in GetCells (new CGSize (width, -1)))
```

Comment:
Possible perf problem could arise because of `GetCells` being implemented via `IEnumerable` + `yield return`. We'll have one boxed allocation for this enumeration and another allocation for the iterator. This method is called in a loop.

Reply:
We know the size of the collection, so we can just return an Array of `CellPos` and then we would be allocationless.

Reply:
👍 will change it